cmd/cfssl/cfssl_test.go

```diff
@@ -1 +1,480 @@
 package main
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/cloudflare/cfssl/api"
+	"github.com/cloudflare/cfssl/auth"
+	"github.com/cloudflare/cfssl/certinfo"
+	"github.com/cloudflare/cfssl/csr"
+	"github.com/cloudflare/cfssl/helpers/testsuite"
+	"github.com/cloudflare/cfssl/info"
+	"github.com/cloudflare/cfssl/scan"
+)
+
+// CFSSL local server to start up
+var server *testsuite.CFSSLServer
+
+// We will test on this address and port.
+// Be sure that these are free or the test will fail.
+var addressToTest = "127.0.0.1"
+var portToTest = 8888
+var postStr = fmt.Sprintf("http://%s:%s/api/v1/cfssl/", addressToTest,
+	strconv.Itoa(portToTest))
+
+var hosts = []byte(`{"hosts":["www.example.com"],
+                     "names":[{"C":"US", "ST":"California", 
+                       "L":"San Francisco", "O":"example.com"}],
+                     "CN": "www.example.com"}`)
+
+var newcertHosts = []byte(`{"hosts":["www.example.com"],
+                     "names":[{"C":"US", "ST":"California", 
+                       "L":"San Francisco", "O":"example.com"}],
+                     "CN": "www.example.com", 
+                     "key":{"algo":"rsa","size":2048}}`)
+
+var requestStr = []byte(`{ "request": ` + string(hosts) + ` }`)
+
+var certRequest = []byte(`{"certificate_request":
```

Comment:
For my RO API test suite I put the certificate in the testdata/ folder. That could work here as well.